src/snyk/cli/services/cliService.ts

```diff
@@ -66,4 +85,26 @@ export abstract class CliService<CliResult> extends AnalysisStatusProvider {
 
     return args;
   }
+
+  public async isChecksumCorrect(cliPath: string): Promise<boolean> {
+    if (this.verifiedChecksumCorrect !== undefined) {
```

Comment:
any reason we're strongly checking against `undefined`?

Reply:
It checks whether the property was set. The property can be set to false, which would fail the early return and execute the verification code on every function call. This is not the behaviour we need here. Does it make sense?